numpy/core/src/private/npy_partition.h.src

```diff
@@ -92,10 +93,12 @@ static NPY_INLINE PyArray_PartitionFunc *
 get_partition_func(int type, NPY_SELECTKIND which)
 {
     npy_intp i;
+    npy_intp ntypes = ARRAY_SIZE(_part_map);
+
     if (which >= NPY_NSELECTS) {
         return NULL;
     }
-    for (i = 0; i < sizeof(_part_map)/sizeof(_part_map[0]); i++) {
+    for (i = 0; i < ntypes; i++) {
```

Comment:
Could consider using `ARRAY_SIZE` in the loop directly

Reply:
That brings us back to the signed-unsigned comparison problem. Assignment is OK and the compiler will optimize things away in any case.

Reply:
Oh right, good point